src/ImageSharp/Colors/Color.BulkOperations.cs

```diff
@@ -0,0 +1,238 @@
+// <copyright file="Color.BulkOperations.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp
+{
+    using System;
+    using System.Numerics;
+    using System.Runtime.CompilerServices;
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Conains the definition of <see cref="BulkOperations"/>
+    /// </content>
+    public partial struct Color
+    {
+        /// <summary>
+        /// <see cref="BulkPixelOperations{TColor}"/> implementation optimized for <see cref="Color"/>.
+        /// </summary>
+        internal class BulkOperations : BulkPixelOperations<Color>
+        {
+            /// <summary>
+            /// SIMD optimized bulk implementation of <see cref="IPixel.PackFromVector4(Vector4)"/>
+            /// that works only with `count` divisible by <see cref="Vector{UInt32}.Count"/>.
+            /// </summary>
+            /// <param name="sourceColors">The <see cref="BufferPointer{T}"/> to the source colors.</param>
+            /// <param name="destVectors">The <see cref="BufferPointer{T}"/> to the dstination vectors.</param>
+            /// <param name="count">The number of pixels to convert.</param>
+            /// <remarks>
+            /// Implementation adapted from:
+            /// <see>
+            ///     <cref>http://stackoverflow.com/a/5362789</cref>
+            /// </see>
+            /// </remarks>
+            internal static unsafe void ToVector4SimdAligned(
+                BufferPointer<Color> sourceColors,
+                BufferPointer<Vector4> destVectors,
+                int count)
+            {
+                int vecSize = Vector<uint>.Count;
+
+                DebugGuard.IsTrue(
+                    count % vecSize == 0,
+                    nameof(count),
+                    "Argument 'count' should divisible by Vector<uint>.Count!");
+
+                Vector<float> bVec = new Vector<float>(256.0f / 255.0f);
+                Vector<float> magicFloat = new Vector<float>(32768.0f);
```

Comment:
To be honest, I don't understand it either :D 
Here's what's happened: Tried the code in the arcticle -> works. Optimized it with SIMD -> still works.